datalad/support/repo.py

```diff
@@ -88,26 +88,30 @@ def _flyweight_id_from_args(cls, *args, **kwargs):
         """
         pass
 
-    def _flyweight_invalid(cls, id):
-        """determines whether or not an instance with `id` became invalid and
-        therefore has to be instantiated again.
-
-        Subclasses can implement this method to provide an additional condition
-        on when to create a new instance besides there is none yet.
-
-        Parameter
-        ---------
-        id: hashable
-          ID of the requested instance
-
-        Returns
-        -------
-        bool
-          whether to consider an existing instance with that ID invalid and
-          therefore create a new instance. Default implementation always returns
-          False.
-        """
-        return False
+    #       TODO: - We might want to remove the classmethod from Flyweight altogether and replace by an
+    #             requirement to implement an actual method, since the purpose of it is actually about a
+    #             particular, existing instance
+    #             - Done. But update docs!
+    # def _flyweight_invalid(cls, id):
+    #     """determines whether or not an instance with `id` became invalid and
+    #     therefore has to be instantiated again.
+    #
+    #     Subclasses can implement this method to provide an additional condition
+    #     on when to create a new instance besides there is none yet.
+    #
+    #     Parameter
+    #     ---------
+    #     id: hashable
+    #       ID of the requested instance
+    #
+    #     Returns
+    #     -------
+    #     bool
+    #       whether to consider an existing instance with that ID invalid and
+    #       therefore create a new instance. Default implementation always returns
+    #       False.
+    #     """
+    #     return False
```

Comment:
Is there value in keeping this around?

Reply:
Only in order to base a rephrased doc on it regarding the actual method requirement, that is the replacement for that classmethod. :-)